02-peer-protocol.md

```diff
@@ -252,6 +272,7 @@ are not valid secp256k1 pubkeys in compressed format.
   - the funder's amount for the initial commitment transaction is not sufficient for full [fee payment](03-transactions.md#fee-payment).
   - both `to_local` and `to_remote` amounts for the initial commitment transaction are less than or equal to `channel_reserve_satoshis` (see [BOLT 3](03-transactions.md#commitment-transaction-outputs)).
   - `funding_satoshis` is greater than or equal to 2^24 and the receiver does not support `option_support_large_channel`. 
+  - It supports `channel_type`, `channel_type` was set, and the `type` is not suitable.
```

Comment:
" - `channel type` is set to a type containing a feature which was note negotiated" ?

Reply:
You don't *need* to do additional checks: having "MUST" here (above) implies you should be testing features.

It's weird, but you should only be policing your own options, *not* theirs.  So if they offer you something you understand, you can take it.

Reply:
> Related comment: [lightningnetwork/lnd#5373 (comment)](https://github.com/lightningnetwork/lnd/pull/5373#issuecomment-908754143)
> 
> If we added a feature bit for something like upfront shutdown (that modified the chan negotiation messages to add new behavior), why wouldn't we add one here for something that revamps the channel type selection negotiation entirely?

This simply allows downgrade from "all the features we negotiated".  If we want to make this required for some new feature we add, that's cool (I think it should be for stuff like option_simplified_commitment, which you might want to support, but not prefer), but we can add that when we add the feature.

I think it's gonna be moot since everyone wants this and in 12 months it'll be compulsory anyway.... (hope)!